src/snyk/common/services/downloadService.ts

```diff
@@ -111,37 +110,24 @@ export class DownloadService {
     return true;
   }
 
-  private async setLastLsUpdateDateAndChecksum(checksum: Checksum): Promise<void> {
-    await this.extensionContext.updateGlobalStateValue(MEMENTO_LS_LAST_UPDATE_DATE, Date.now());
-    await this.extensionContext.updateGlobalStateValue(MEMENTO_LS_CHECKSUM, checksum.checksum);
-  }
-
-  private async setCurrentLspVersion(): Promise<void> {
-    await this.extensionContext.updateGlobalStateValue(MEMENTO_LS_PROTOCOL_VERSION, PROTOCOL_VERSION);
-  }
-
-  private isFourDaysPassedSinceLastLsUpdate(): boolean {
-    const lastUpdateDate = this.getLastLsUpdateDate();
-    if (!lastUpdateDate) {
-      throw new Error('Last update date is not known.');
-    }
-    return Date.now() - lastUpdateDate > this.fourDaysInMs;
+  private async setCliChecksum(checksum: Checksum): Promise<void> {
+    await this.extensionContext.updateGlobalStateValue(MEMENTO_CLI_CHECKSUM, checksum.checksum);
   }
 
-  private hasLspVersionUpdated(): boolean {
-    const currentProtoclVersion = this.getLsProtocolVersion();
-    return currentProtoclVersion != PROTOCOL_VERSION;
+  private async setCliVersion(cliVersion: string): Promise<void> {
+    await this.extensionContext.updateGlobalStateValue(MEMENTO_CLI_VERSION, cliVersion);
   }
 
-  private getLastLsUpdateDate() {
-    return this.extensionContext.getGlobalStateValue<number>(MEMENTO_LS_LAST_UPDATE_DATE);
+  private hasCliVersionUpdated(cliVersion: string): boolean {
+    const currentVersion = this.getCliVersion();
+    return currentVersion != cliVersion;
   }
 
-  private getLsProtocolVersion() {
-    return this.extensionContext.getGlobalStateValue<number>(MEMENTO_LS_PROTOCOL_VERSION);
+  private getCliVersion(): string | undefined {
+    return this.extensionContext.getGlobalStateValue<string>(MEMENTO_CLI_VERSION);
```

Comment:
nitpick(non-blocking): We should probably clean up the old mementos

Reply:
question: where is the protocol version (of LS) saved and compared against the requiredProtocol Version? Has this change an impact on that?